DataDog/datadog-cdk-constructs
# chore: Add snapshot test for Go stacks

### What does this PR do?
Add a Go stack to snapshot tests, so that the test will compare the snapshots generated after vs before, to check if a PR causes unexpected changes on the stack to be deployed.

The stack was copied from `examples/go-stack`. I had to copy it because I don't think it's possible to just reference the existing stack.



### Motivation
Adding more tests to make it safer to deploy changes.



### Testing Guidelines
To ensure the example stack uses the locally built Go package instead of the production package, I:
1. Add a line to [`applyEnvVariables()`](https://github.com/DataDog/datadog-cdk-constructs/blob/main/src/env.ts#L122) in `env.ts`:
```
lam.addEnvironment("Ah", "Oh");
```
so an extra environment variable will be added to the Lambda function.

2. Run `yarn build`
3. Run the snapshot test `aws-vault exec sso-serverless-sandbox-account-admin -- scripts/run_integration_tests.sh`

Result:
The test failed. The generated snapshot has the new environment variable, which captures the unintended change.



### Additional Notes



### Types of Changes


### Check all that apply
